lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -997,12 +997,18 @@ def payment_details(payment_method)
         when String
           token_type_and_details(payment_method)
         else
-          type = payment_method.respond_to?(:source) ? :network_token : :credit
+          type = network_token?(payment_method) ? :network_token : :credit
```

Comment:
is this better than...?
```
when NetworkTokenizationCreditCard
  type = :network_token
else
  type = :credit
```

Reply:
It was like that before https://github.com/activemerchant/active_merchant/commit/51c1f2d0333c53b89a8fb12189838694e5b09e6b and I undertand that https://github.com/activemerchant/active_merchant/commit/51c1f2d0333c53b89a8fb12189838694e5b09e6b is based on the fact that the `active_merchant` gem does not specify strictly what classes are meant to be used. In other words, the gem is _ducked-type_, ie its _public interfaces [...] are not tied to any specific class_.